### PR TITLE
[FE/API]: update api style-guide, add fe style-guide

### PR DESCRIPTION
**Pull request by vadim-ilchenko**

### CODE REVIEW

**Review by Anna-Tkachenko — APPROVED** [review]
⭐️

File: mate-api.md [hunk 1]
```diff
@@ -324,6 +328,80 @@ type CourseUser {
...
+  // ... types go here
+  >({
+    Handler: GetChatLessonUseCase,
+    auth: true,
+    mapOptions: (args, parent) => ({
+      chatId: chat.id,
+    }),
+  })
+
+export class getCareerTestQuestionsResolver {}
```
Comment (VinnieJ-2k20) [inline mate-api.md]: See next example :harold:
```suggestion
export class GetCareerTestQuestionsResolver {}
```

**Review by VinnieJ-2k20 — APPROVED** [review]
🔥

**Review by G-YaSha — APPROVED** [review]
👍

**Review by ivan-kadykalo — APPROVED** [review]

File: mate-api.md (hunk 1 above)
Comment (G-YaSha) [reply mate-api.md]: cc @ivan-kadykalo Seems it is your resolver. Pay attention to the resolvers name convention

File: mate-api.md [hunk 2]
```diff
@@ -450,4 +528,94 @@ if (User.typingSpeedTests?.length > 0) {
...
+
+It is mandatory to understand the major difference between seeders and migrations:
+- **seeders** are used to modify the database content (adding or removing entries)
+- **migrations** are used to modify the database structure (adding new table/column, managing constraints, etc)
+
+Besides seeders are not executed in CI/Tests while migrations are
+
+#### 4.2. Follow naming template for seeders and migrations
+
+The template is: `{action}-{object}-{subject}` in an imperative form
```
Comment (G-YaSha) [inline mate-api.md:545]: 🔥🔥🔥🔥🔥

File: mate-api.md [hunk 3]
```diff
@@ -450,4 +528,94 @@ if (User.typingSpeedTests?.length > 0) {
...
+          ]
+      }
+    })))
+
+// ✅ good
+await queryInterface.addColumn(
+  'users',
+  'student_status',
+  {
+    type: Sequelize.ENUM(
```
Comment (G-YaSha) [inline mate-api.md:613]: Didn't know about such an approach. Thanks👍

File: mate-api.md (hunk 1 above)
Comment (vadim-ilchenko) [reply mate-api.md]: :haroldface:

File: mate-api.md [hunk 4]
```diff
@@ -450,4 +528,94 @@ if (User.typingSpeedTests?.length > 0) {
   // ...
 }
 
+```
+
+4\. Database
```
Comment (G-YaSha) [inline mate-api.md:533]: Wdyt about adding convention for indexes/constraints names? I was thinking about adding this point some time ago. I didn't add this point, cause it's a rather rare case when people name indexes/constraints in a wrong way

Here is the example of such case: https://github.com/mate-academy/website/pull/15822#discussion_r1041430344

And here is an example of some rules for naming constraints/indexes in `PostgreSQL`: https://gist.github.com/popravich/d6816ef1653329fb1745

File: mate-frontend.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+```scss
+.listItem {
+  margin-bottom: rem-calc(16);
+}
+```
+
+❌ bad
+
+Foundation provides set of atomic indents classes but they apply `!important` flag and are inconsystem with our system
```
Comment (jstmpelowycz) [inline mate-frontend.md:113]: inconsystem? perhaps, incompatible?

File: mate-frontend.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    // content
+  </li>
+</ul>
+```
+
+✅ good
+```jsx
+<ul>
+  // applies margin-bottom: 1rem;
+  <li className="mb-16">
```
Comment (jstmpelowycz) [inline mate-frontend.md:128]: wow! have never used it, thanks!

File: mate-frontend.md (hunk 5 above)
Comment (vadim-ilchenko) [reply mate-frontend.md:113]: oh, thank you, it's a kind of wrong-fixed typo 💇

**Pull request merged**